### DISCUSSION TITLE
Anyway to get metric names in events?

### DISCUSSION THREAD

**Discussion by lozbrown** [kedacore/keda#6417] [Q&A / Need Help]
We have some metrics like so 

>       - type: prometheus
>         name: queuedQueries
>         metricType: Value
>         metadata:
>           # Required fields:
>           serverAddress: https://grafana.someplace/api/datasources/proxy/312
>           query: avg_over_time(trino_execution_name_QueryManager_QueuedQueries{pipelineBranch="somebranch",namespace="trino"}[2m]) - avg_over_time(trino_execution_resourcegroups_type_InternalResourceGroup_name_global_limited_QueuedQueries{pipelineBranch="somebranch",namespace="trino"}[2m]) #This is to exclude the noisy resource groups.  Note: query must return a vector/scalar single element response
>           threshold: '5'
>           activationThreshold: '0'
>           # Optional fields:
>           name: queuedQueries
>           customHeaders: X-Grafana-Org-Id=97 # Optional. Custom headers to include in query. In case of auth header, use the custom authentication or relevant authModes.
>           queryParameters: datasourceId=312
>           unsafeSsl: "true" #  Default is `false`, Used for skipping certificate check when having self-signed certs for Prometheus endpoint
>           authModes: "bearer"
>         authenticationRef:
>           name: trino-prom-creds

these seem to produce kubernetes events that we export ti the dashboard like this 

> New size: 3; reason: external metric s2 above target

IS there anyway to make these events have a meaningful name instead of s1, s2 etc

**Comment by lozbrown** [comment]
in Argo i see scaled object and hpa attached to that

the keda scaled object shows the metric names in the manifest but the hps shows the s1,s2 etc

**Comment by nadav-tk** [comment]
We are suffering from it too.
Prometheus metrics show up as "s0-prometheus", "s1-prometheus" etc on the generated HPA, and it is difficult to tell which is which.
It would really help to see the metric name on the HPA, either standalone or concatanated to the "s[n]-prometheus
